dl/translation: metrics for input throughput

andrwng opened the pull request:
Adds some quick-and-dirty metrics for throughput of incoming Kafka data meant only for use in debugging.

E.g. this should not be used for usage based billing, given it doesn't account for commits into Iceberg, doesn't account for errors in the translation path, and gets reset when partitions are removed or shutdown.

<!--
See https://github.com/redpanda-data/redpanda/blob/dev/CONTRIBUTING.md#pull-request-body
for more details and examples of what is expected in a PR body.

Content in this top section is REQUIRED. Describe, in plain language, the motivation
behind the change (bug fix, feature, improvement) in this PR and how the included
commits address it.

Add the GitHub keyword `Fixes` to link to bug(s) this PR will fix, e.g.
  Fixes #ISSUE-NUMBER, Fixes #ISSUE-NUMBER, ...

If this PR is a backport, link to the original with `Backport of PR`, e.g.
  Backport of PR #PR-NUMBER
-->

## Backports Required

<!-- Checking at least one of the checkboxes is REQUIRED if this PR is not a backport. -->

- [ ] none - not a bug fix
- [ ] none - this is a backport
- [ ] none - issue does not exist in previous branches
- [ ] none - papercut/not impactful enough to backport
- [x] v25.1.x
- [ ] v24.3.x
- [ ] v24.2.x
- [ ] v24.1.x

## Release Notes

* none
<!--
If the changes in this PR do not need to be mentioned in the release
notes, then don't add a sub-section and simply list `none`, e.g.

* none

Otherwise, adding a sub-section or `none` is REQUIRED if the PR is not a backport PR.
If this is a backport PR, adding contents to this section will override
the release notes section inherited from the original PR to dev.

Add one or more of the sub-sections with a short description bullet
point of the change, e.g.

### Bug Fixes

* Short description of the bug fix if this is a PR to `dev` branch.

### Features

* Short description of the feature. Explain how to configure.

### Improvements

* Short description of how this PR improves existing behavior.

-->

ztlpn commented on src/v/datalake/translation/translation_probe.h:
```diff
@@ -61,6 +71,13 @@ class translation_probe final {
+
+    // NOTE: the accounting for bytes here is not strictly accurate and should
+    // only be used to get a rough sense for translation throughput.
+    size_t raw_bytes_processed = 0;
+    size_t decompressed_bytes_processed = 0;
+    size_t raw_bytes_translated = 0;
+    size_t decompressed_bytes_translated = 0;
```
nit: `_` prefix for private members

andrwng replied:
Done

andrwng committed:
dl/translation: metrics for input throughput

Adds some quick-and-dirty metrics for throughput of incoming Kafka data
meant only for use in debugging.

E.g. this should not be used for usage based billing, given it doesn't
account for commits into Iceberg, doesn't account for errors in the
translation path, and gets reset when partitions are removed or
shutdown.

andrwng commented:
Force push is a rebase to address a conflict

rockwotj commented:
Also usage based building doesn't use public metrics as that is prone to scape timing/inconsistency

rockwotj commented on src/v/datalake/translation/translation_probe.cc:
```diff
@@ -97,6 +97,77 @@ void translation_probe::register_created_files_metrics() {
+           "raw_bytes_processed",
+           _raw_bytes_processed,
+           sm::description(
+             "Number of raw bytes consumed for translation that may or may not "
```
Might be worth calling out raw means maybe compressed

andrwng replied:
Done

rockwotj commented on src/v/datalake/translation/translation_probe.cc:
```diff
@@ -97,6 +97,77 @@ void translation_probe::register_created_files_metrics() {
+           sm::description(
+             "Raw bytes consumed for translation that were successfully "
+             "translated. Note that translation in to the DLQ is still "
+             "considered success. Only accounts for translation and not the "
```
Maybe: the bytes measured here are for files created before they are committed to the iceberg table

rockwotj commented on src/v/datalake/translation/translation_probe.cc:
```diff
@@ -97,6 +97,77 @@ void translation_probe::register_created_files_metrics() {
+      group_name,
+      {{
+         sm::make_counter(
+           "raw_bytes_processed",
```
I am not sure we need the processed vs success distinction given the cost of metrics, but I guess it's fine 🤷

andrwng replied:
Yea that's fair. Removed the success case since i think processing is more useful to know about for debugging

rockwotj commented on src/v/datalake/translation/translation_probe.cc:
```diff
@@ -97,6 +97,77 @@ void translation_probe::register_created_files_metrics() {
+       },
+       {
+         sm::make_counter(
+           "raw_bytes_translated",
```
I don't love terms like translation leaking into the naming because I don't think it's intuitive for customers (at least at first glance), but I guess we have it in other places so that ship has sailed

andrwng replied:
Hmm... maybe could've been "raw_bytes_consumed_for_parquet" or something? Anyway, I removed the "translated" (success) metric and reworded the processing metric description

vbotbuildovich commented:
/backport v25.1.x

vbotbuildovich commented:
Failed to create a backport PR to v25.1.x branch. I tried:

```
git remote add upstream https://github.com/redpanda-data/redpanda.git
git fetch --all
git checkout -b backport-pr-25592-v25.1.x-516 remotes/upstream/v25.1.x
git cherry-pick -x aaeb8001ad
```

[Workflow run logs](https://github.com/redpanda-data/redpanda/actions/runs/14103904100).
